### PR TITLE
Sort key and filters as a list of columns in list requests

### PR DESCRIPTION
**Pull request by iaroslav-ciupin**
## _Read then delete this section_

_- Make sure to use a concise title for the pull-request._

_- Use #patch, #minor or #major in the pull-request title to bump the corresponding version. Otherwise, the patch version
will be bumped. [More details](https://github.com/marketplace/actions/github-tag-bump)_

# TL;DR
In order to fix SQL injection we will deprecate and restrict existing `sort_by` to just single column and introduce new field `sort_keys` that will accept array of columns with directions.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: protos/flyteidl/admin/common.proto [hunk 1]
```diff
@@ -210,9 +210,13 @@ message ResourceListRequest {
     // +optional
     string filters = 4;
 
-    // Sort ordering.
+    // Sort column ordering.
     // +optional
-    Sort sort_by = 5;
+    // Deprecated.
```
Comment (katrogan) [inline protos/flyteidl/admin/common.proto:215]: nit can we add a comment here to use sort_keys instead?

**Review by katrogan — APPROVED** [review]

**Comment by hamersaw** [comment]
can we fix the generate verification test?

**Comment by iaroslav-ciupin** [comment]
@hamersaw I'm not sure I know how to do this.

File: protos/flyteidl/admin/common.proto [hunk 2]
```diff
@@ -210,9 +210,13 @@ message ResourceListRequest {
...
 
-    // Sort ordering.
+    // Sort column ordering.
     // +optional
-    Sort sort_by = 5;
+    // Deprecated. Use sort_keys instead.
+    Sort sort_by = 5 [deprecated = true];
+
+    // Sort column(s) with direction.
+    repeated Sort sort_keys = 6;
```
Comment (katrogan) [inline protos/flyteidl/admin/common.proto:219]: does this work with http requests in grpc gateway v1? 
I recall we used unstructured string for filters (https://docs.flyte.org/en/latest/concepts/admin.html#adding-request-filters)

>  While this allows for a single implementation for both HTTP and gRPC, an important limitation is that fields mapped to the path pattern cannot be repeated and must have a primitive (non-message) type. Unfortunately this means that repeated string filters cannot use a proper protobuf message.

**Comment by iaroslav-ciupin** [comment]
this change is not needed right now

**Pull request closed without merging**